react/Breadcrumb.tsx

```diff
@@ -1,79 +1,85 @@
 import React, { Fragment, useMemo } from 'react'
-import { Link } from 'vtex.render-runtime'
 import unorm from 'unorm'
-import { IconHome, IconCaret } from 'vtex.store-icons'
+import { Link } from 'vtex.render-runtime'
+import { IconCaret, IconHome } from 'vtex.store-icons'
 
 import styles from './breadcrumb.css'
 
 const LINK_CLASS_NAME = `${styles.link} dib pv1 link ph2 c-muted-2 hover-c-link`
 
 interface Category {
   name: string
-  link: string
+  href: string
 }
+
 interface Props {
   term?: string
   /** Shape [ '/Department' ,'/Department/Category'] */
-  categories: Array<string>
+  categories: string[]
+  categoryTree: Category[]
 }
 
-const makeLink = (string: string) =>
+const makeLink = (str: string) =>
   unorm
-    .nfd(string)
+    .nfd(str)
     .toLowerCase()
     .replace(/[\u0300-\u036f]/g, '')
     .trim()
     .replace(/[-\s]+/g, '-')
 
+const getCategoriesList = (categories: string[]) : Category[] => {
+  const categoriesSorted = categories
+    .slice()
+    .sort((a, b) => a.length - b.length)
+  return categoriesSorted.map(category => {
+    const categoryStripped = category
+      .replace(/^\//, '')
+      .replace(/\/$/, '')
+    const currentCategories = categoryStripped.split('/')
+    const [categoryKey] = currentCategories.reverse()
+    const linkCompletion = currentCategories.length === 1 ? '/d' : ''
+    const href = `/${makeLink(categoryStripped)}${linkCompletion}`
+    return {
+      href,
+      name: categoryKey,
+    }
+  })
+}
+
 /**
  * Breadcrumb Component.
  */
-const Breadcrumb = ({ term, categories }: Props) => {
-
-  const getCategoriesList = (categories:Array<string>) : Array<Category> => {
-    const categoriesSorted = categories
-      .slice()
-      .sort((a, b) => a.length - b.length)
-    return categoriesSorted.map(category => {
-      const categoryStripped = category
-        .replace(/^\//, '')
-        .replace(/\/$/, '')
-      const currentCategories = categoryStripped.split('/')
-      const [categoryKey] = currentCategories.reverse()
-      const linkCompletion = currentCategories.length === 1 ? '/d' : ''
-      const link = makeLink(categoryStripped) + linkCompletion
-      return {
-        name: categoryKey,
-        link: link,
-      }
-    })
-  }
-  const categoriesList = useMemo(() => getCategoriesList(categories), [categories])
+const Breadcrumb = ({ term, categories, categoryTree }: Props) => {
+  const categoriesList = categoryTree 
+    ? categoryTree
+    : useMemo(() => getCategoriesList(categories), [categories])
```

Comment:
hooks can't be used conditionally, you should either **always** use the hook or never use it. I think this could be resolved by always using the `useMemo` hook, but reassign the variable if we receive the `categoryTree` prop.

Reply:
thanks ! I didn't know that